src/index.ts

```diff
@@ -1,151 +1,237 @@
 import { minimize_Powell } from 'optimization-js';
 import { cloneDeep } from 'lodash';
+import {Stimulus, Zeta} from './type';
+import {itemResponseFunction, fisherInformation, normal} from './utils';
+import seedrandom from 'seedrandom';
 
-export type Zeta = { a: number; b: number; c: number; d: number };
+export const abilityPrior = normal();
 
-export interface Stimulus {
-  difficulty: number;
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  [key: string]: any;
+export interface CatInput {
+  method?: string;
+  itemSelect?: string;
+  nStartItems?: number;
+  startSelect?: string;
+  theta?: number;
+  minTheta?: number;
+  maxTheta?: number;
+  prior?: number[][];
+  randomSeed?: string | null
 }
 
-/**
- * calculates the probability that someone with a given ability level theta will answer correctly an item. Uses the 4 parameters logistic model
- * @param theta - ability estimate
- * @param zeta - item params
- * @returns {number} the probability
- */
-export const itemResponseFunction = (theta: number, zeta: Zeta) => {
-  return zeta.c + (zeta.d - zeta.c) / (1 + Math.exp(-zeta.a * (theta - zeta.b)));
-};
-
-/**
- * a 3PL Fisher information function
- * @param theta - ability estimate
- * @param zeta - item params
- * @returns {number} - the expected value of the observed information
- */
-export const fisherInformation = (theta: number, zeta: Zeta) => {
-  const p = itemResponseFunction(theta, zeta);
-  const q = 1 - p;
-  return Math.pow(zeta.a, 2) * (q / p) * (Math.pow(p - zeta.c, 2) / Math.pow(1 - zeta.c, 2));
-};
-
-/**
- * return a Gaussian distribution within a given range
- * @param mean
- * @param stdDev
- * @param min
- * @param max
- * @param stepSize - the quantization (step size) of the internal table, default = 0.1
- * @returns {Array<[number, number]>} - a normal distribution
- */
-export const normal = (mean = 0, stdDev = 1, min = -4, max = 4, stepSize = 0.1) => {
-  const distribution = [];
-  for (let i = min; i <= max; i += stepSize) {
-    distribution.push([i, y(i)]);
-  }
-  return distribution;
-
-  function y(x: number) {
-    return (1 / (Math.sqrt(2 * Math.PI) * stdDev)) * Math.exp(-Math.pow(x - mean, 2) / (2 * Math.pow(stdDev, 2)));
-  }
-};
+export class Cat {
+  public method: string;
+  public itemSelect: string;
+  public minTheta: number;
+  public maxTheta: number;
+  public prior: number[][];
+  private readonly _zetas: Zeta[];
+  private readonly _answers: (0|1)[];
+  private _nItems: number;
+  private _theta: number;
+  private _seMeasurement: number;
+  public nStartItems: number;
+  public startSelect: string;
+  private readonly _rng: (...args:any[]) => any;
```

Comment:
```suggestion
  private readonly _rng: typeof seedrandom;
```

Using `typeof` here might be better. It's the most specific we can be so it tells TypeScript exactly what we want. And I think it will avoid the warnings from the GitHub actions.

Reply:
This suggestion doesn't work. It says ' Type 'PRNG' is missing the following properties from type 'seedrandom': alea, Alea, tychei, Tychei, and 8 more.' Any idea?

Reply:
Maybe `typeof seedrandom()` with the parentheses

Reply:
still not working... we can chat async about this tomorrow

Reply:
Sorry for the delayed response. I think maybe the recommended way to handle this is with `ReturnType<Type>` (see https://www.typescriptlang.org/docs/handbook/utility-types.html#returntypetype)

So perhaps it should read
```
private readonly _rng: ReturnType<seedrandom>;
```